examples/run_FS.py

```diff
@@ -32,148 +32,163 @@
-    # should work same as line below and be more effic
-    #data = raw_data.view(complex128)
-    data[::] = complex128(raw_data[0::2]+1j*raw_data[1::2])
-    print("COMPLEX DATA ARRAY LENGTH:",shape(data)[0])
-    print("RAW DATA ARRAY LENGTH:",shape(raw_data)[0])
-    dataPoints = float(shape(data)[0])
-    time_axis = linspace(0.0,nEchoes*nPhaseSteps*acq_time*1e-3,dataPoints)
-    data = nddata(array(data),'t')
-    data.setaxis('t',time_axis).set_units('t','s')
-    data.name('signal')
-    if index == 0:
-        field_sweep = ndshape([len(field_axis),len(time_axis)],['field','t']).alloc(dtype=complex128)
-        field_sweep.setaxis('field',field_axis).set_units('field','G')
-        field_sweep.setaxis('t',time_axis).set_units('t','s')
-    field_sweep['field',index] = data
-SpinCore_pp.stopBoard();
-print("EXITING...")
-print("\n*** *** ***\n")
-save_file = True
-while save_file:
+nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+#}}}
+total_pts = nPoints * nPhaseSteps
+assert total_pts < 2 ** 14, (
+    "You are trying to acquire %d points (too many points) -- either change SW or acq time so nPoints x nPhaseSteps is less than 16384"
+    % total_pts
+)
+with xepr() as x_server:
+        first_B0 = x_server.set_field(field_axis[0])
+        time.sleep(3.0)
+        carrierFreq_MHz = config_dict["gamma_eff_MHz_G"] * first_B0
+        sweep_data = run_spin_echo(
+            nScans=config_dict["nScans"],
+            indirect_idx=0,
+            indirect_len=len(field_axis),
+            adcOffset=config_dict["adc_offset"],
+            carrierFreq_MHz=carrierFreq_MHz,
+            nPoints=nPoints,
+            nEchoes=config_dict["nEchoes"],
+            p90_us=config_dict["p90_us"],
+            repetition=config_dict["repetition_us"],
+            tau_us=config_dict["tau_us"],
+            SW_kHz=config_dict["SW_kHz"],
+            output_name=filename,
+            indirect_fields=("Field", "carrierFreq"),
+            ret_data=None,
+        )
+        myfreqs_fields = sweep_data.getaxis("indirect")
+        myfreqs_fields[0]["Field"] = first_B0
+        myfreqs_fields[0]["carrierFreq"] = config_dict["carrierFreq_MHz"]
+        for B0_index, desired_B0 in enumerate(field_axis[1:]):
+            true_B0 = x_server.set_field(desired_B0)
+            logging.info("My field in G is %f" % true_B0)
+            time.sleep(3.0)
+            new_carrierFreq_MHz = config_dict["gamma_eff_MHz_G"] * true_B0
+            myfreqs_fields[B0_index + 1]["Field"] = true_B0
+            myfreqs_fields[B0_index + 1]["carrierFreq"] = new_carrierFreq_MHz
+            logging.info("My frequency in MHz is", new_carrierFreq_MHz)
+            run_spin_echo(
+                nScans=config_dict["nScans"],
+                indirect_idx=B0_index + 1,
+                indirect_len=len(field_axis),
+                adcOffset=config_dict["adc_offset"],
+                carrierFreq_MHz=new_carrierFreq_MHz,
```

Comment:
again as the field changes we need to change the carrier frequency to remain on resonance using the effective gamma